pkg/jobconfig/files.go

```diff
@@ -271,11 +272,11 @@ func readFromFile(path string) (*prowconfig.JobConfig, error) {
 	return jobConfig, nil
 }
 
-// Given a JobConfig and a target directory, write the Prow job configuration
+// WriteToDir takes a JobConfig and a target directory, and writes the Prow job configuration
 // into files in that directory. Jobs are sharded by branch and by type. If
 // target files already exist and contain Prow job configuration, the jobs will
-// be merged.
-func WriteToDir(jobDir, org, repo string, jobConfig *prowconfig.JobConfig) error {
+// be merged. Jobs will be pruned based on the provided generatedLabels
+func WriteToDir(jobDir, org, repo string, jobConfig *prowconfig.JobConfig, generatedLabels ...string) error {
 	allJobs := sets.String{}
 	files := map[string]*prowconfig.JobConfig{}
 	key := fmt.Sprintf("%s/%s", org, repo)
```

Comment:
Was it not possible to *add* the newly-generated labels to the jobs in the following loops, instead of requiring the caller to remember to do so in every case?

Reply:
Note that this is probably true of `prowgen` as well.  I mentioned in my other PRs that we can rethink pruning now that there isn't a round trip to the file system as there used to be and we have old and new jobs conveniently in memory.

I was making a large enough change at the time, so I left it out, but we can consider that if we're making changes here.